bgpd/bgp_clist.c

```diff
@@ -1026,6 +1026,31 @@ struct lcommunity *lcommunity_list_match_delete(struct lcommunity *lcom,
 	return lcom;
 }
 
+/* Helper to check if every octet do not exceed UINT_MAX */
+static int lcommunity_list_valid(const char *str)
+{
+	const char *delimiter = ":";
+	char *community = strdup(str);
+	char *ptr = NULL;
+	int octets = 0;
+
+	for (char *octet = strtok_r(community, delimiter, &ptr); octet;
+	     octet = strtok_r(NULL, delimiter, &ptr)) {
+		if (strtoul(octet, NULL, 10) > UINT_MAX)
+			return 0;
+
+		if (strlen(octet) == 0)
+			return 0;
+
+		octets++;
+	}
```

Comment:
FRR has string splitting stuff in lib:
```
char **splits;
int num;
frrstr_split(str, ':', &splits, &num);

for (int i = 0; i < num; i++) {
   ... strtoul range checks here ...
  XFREE(MTYPE_TMP, splits[i]);
}

XFREE(MTYPE_TMP, splits);

... check argc etc ...
```
bit simpler than what is done here imo

Reply:
I'll rewrite using this pattern.

Reply:
addressed :)